### PR TITLE
153 - buttons - new functional colour variants

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ComponentButtonsContained.kt [hunk 1]
```diff
@@ -17,69 +17,82 @@ import androidx.compose.ui.Modifier
...
     DarkSurface {
-        ContainedButtonsAppearanceForced(OdsDisplayAppearance.ON_DARK, hasPrimaryColor = hasPrimaryColor)
+        ContainedButtonsAppearanceForced(OdsDisplaySurface.Dark, functionalType = functionalType)
     }
 }
 
+
 @Composable
-private fun ContainedButtonsAppearanceForced(displayAppearance: OdsDisplayAppearance, hasPrimaryColor: Boolean) {
+private fun ContainedButtonsAppearanceForced(displaySurface: OdsDisplaySurface, functionalType: OdsButtonFunctionalType) {
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/buttons/ComponentButtonsContained.kt:62]: You're right!

File: demo/src/main/java/com/orange/ods/demo/ui/components/dialogs/ComponentDialogs.kt [hunk 2]
```diff
@@ -49,7 +50,7 @@ fun ComponentDialogsContent() {
     OdsButton(
         modifier = Modifier.fullWidthButton(),
         text = stringResource(id = R.string.component_dialogs_open_dialog),
-        hasPrimaryColor = true,
+        functionalType = OdsButtonFunctionalType.Primary,
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/dialogs/ComponentDialogs.kt:53]: But I'm not sure cause, for example, TextButton cannot be green or red but only black or orange...

File: lib/src/main/java/com/orange/ods/compose/component/button/OdsButton.kt [hunk 3]
```diff
@@ -25,10 +25,32 @@ import androidx.compose.runtime.Composable
...
+import com.orange.ods.compose.theme.Positive100
+import com.orange.ods.compose.theme.White100
 import com.orange.ods.compose.theme.odsDarkThemeColors
 import com.orange.ods.compose.theme.odsLightThemeColors
 
+/**
+ * Specifying an [OdsButtonFunctionalType] allow to display a button with functional colors.
+ * Note that `Positive`and `Negative` values are for contained buttons only.
+ */
+enum class OdsButtonFunctionalType {
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/button/OdsButton.kt:40]: So `OdsButtonFunctionalType` isn't correct at all... But I don't know how to name it...

[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/button/OdsButton.kt:40]: `OdsButtonStyle`?

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**